community/kernel/src/main/java/org/neo4j/kernel/impl/store/ShortArray.java

```diff
@@ -767,7 +767,12 @@ public int getRequiredBits( long value )
 
     public static ShortArray typeOf( byte typeId )
     {
-        return ShortArray.values()[typeId-1];
+        ShortArray[] values = ShortArray.values();
+        if ( typeId <= 0 | typeId > values.length )
+        {
+            return null;
```

Comment:
Looking at it again, I'm not sure this change is needed at all. All paths leading here have already successfully copied the relevant data into PropertyBlocks and byte arrays. I'll revert this change.